llm/generate_cuda.go

```diff
@@ -0,0 +1,3 @@
+package llm
+
+//go:generate bash -c "nvcc --forward-unknown-to-host-compiler -use_fast_math -arch=native -DGGML_CUDA_DMMV_X=32 -DGGML_CUDA_MMV_Y=1 -DK_QUANTS_PER_ITERATION=2 -I. -I./examples -O3 -std=c++11 -fPIC -DNDEBUG -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wno-multichar -pthread -march=native -mtune=native -DGGML_USE_K_QUANTS -DGGML_USE_CUBLAS -I/usr/local/cuda/include -I/opt/cuda/include -I/targets/x86_64-linux/include -Wno-pedantic -c ggml-cuda.cu -o ggml-cuda.syso"
```

Comment:
I think there's two ways this should be handled. Either the object file is checked in and should be updated in `update-llama-cpp.sh` or it should be built when needed. I would prefer the latter.

Based on the command line, it might be possible to trick `cgo` to use `nvcc` instead of the default C compiler by overriding `CC`. That way you can passing CFLAGS, CPPFLAGS, LDFLAGS just like any other `cgo` directive. This might have side effects it needs to be tested extensively